### ISSUE TITLE
plan's output for fuzzy version string is confusing/misleading

### ISSUE THREAD

**Issue by davidxia** [terraform-providers/terraform-provider-google#3196]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment
* If an issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

If you specify a fuzzy `node_version`, `terraform plan`'s output is misleading. It'll always say it'll change the cluster's version from some specific GKE version to the fuzzy version you specified. But when you `apply`, it'll be a no-op (assuming your GKE nodes are on the latest version that matches the fuzzy version).

`plan`'s output is thus misleading and confusing. Can terraform be smarter about when a fuzzy version will have actual changes?

### Terraform Version

0.11.10

### Affected Resource(s)

* google_container_cluster

### Expected Behavior

`terraform plan` should say no changes will be made to infra. `terraform apply` is a no-op.

### Actual Behavior

`terraform plan` output looks like it'll change infra. But `apply` is a no-op.

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

~ module.asia-east1-k33l.google_container_cluster.spotify_gke_cluster
      node_pool.0.version: "1.11.7-gke.6" => "1.11"

  ~ module.europe-west1-s41l.google_container_cluster.spotify_gke_cluster
      node_pool.0.version: "1.11.7-gke.6" => "1.11"

  ~ module.production-asia-east1.google_container_cluster.spotify_gke_cluster
      node_pool.0.version: "1.11.7-gke.6" => "1.11"

  ~ module.testing-europe-west1.google_container_cluster.spotify_gke_cluster
      node_pool.0.version: "1.11.7-gke.6" => "1.11"

  ~ module.us-central1-b04t.google_container_cluster.spotify_gke_cluster
      node_pool.0.version: "1.11.7-gke.6" => "1.11"
Plan: 0 to add, 6 to change, 0 to destroy.

------------------------------------------------------------------------

This plan was saved to: tf.plan

To perform exactly these actions, run the following command to apply:
    terraform apply "tf.plan"
```

### Steps to Reproduce

1. Use terraform to create a GKE cluster
1. specify a fuzzy `node_version` like `1.11`
1. run `terraform plan` and notice the output

### References

* related to #3155

**Comment by davidxia** [comment]
Or just state this kind of behavior is expected and link to this [GKE doc]( https://cloud.google.com/kubernetes-engine/versioning-and-upgrades#specifying_cluster_version) in TF docs?

**Comment by davidxia** [comment]
A little more context and something new I noticed today.

My team has 7 GKE clusters we manage via TF. Three have over 50 nodes. Fuzzy versions cause TF to make upgrade requests to the GCP API (PUT HTTP requests with the `1.11` version in the payload). This causes the GCP API to reconcile the cluster to try to match the version. The upgrade request might be a no-op if the GKE version is already the latest, or there might be an actual upgrade.

This behavior caused by fuzzy versions causes our `terraform apply` to almost always timeout. The large GKE clusters' reconciliation takes a long time, TF keeps polling, and maybe because of [this now-resolved bug](https://github.com/GoogleCloudPlatform/magic-modules/pull/1451) eventually times out. We've increased the timeout, but this approach is just a bandaid.

We're going to switch to using specific versions for now to avoid these downsides.

Disadvantages of fuzzy versions:

• potentially upgrading masters and nodes any time a master build runs
• misleading TF plan output

Advantages:

• less prone to break when GKE versions become unavailable
• using more recent versions of GKE
